### PR TITLE
Allow users to access VariantDebugInfo by setting a flag

### PR DESCRIPTION
**Pull request by e7mac**
PRD: https://leanplum.atlassian.net/wiki/spaces/PRODDEV/pages/548503556/Tinder+Expose+granular+information+on+variables+and+A+B+tests+to+SDK

We allow a flag to be set in "includeVariantDebugInfo" that retrieves content assignment details from the server. We then allow access to this object through Leanplum.variantDebugInfo

### CODE REVIEW

**Comment by e7mac** [comment]
test this please

**Comment by e7mac** [comment]
test this please

File: AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java [hunk 1]
```diff
@@ -2043,6 +2063,17 @@ public static Object objectForKeyPathComponents(Object[] pathComponents) {
     return messages;
   }
 
+  /**
+   * Details about the variable assignments on the server.
+   */
+  public static Map<String, Object> variantDebugInfo() {
+    Map<String, Object> variantDebugInfo = VarCache.getVariantDebugInfo();
+    if (variantDebugInfo == null) {
```
Comment (rossenassenov) [inline AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java]: we wouldn't need this check if getVariantDebugInfo always returns non-null

File: AndroidSDKCore/src/main/java/com/leanplum/internal/VarCache.java [hunk 2]
```diff
@@ -322,6 +323,14 @@ public static boolean hasReceivedDiffs() {
     return hasReceivedDiffs;
   }
 
+  public static Map<String, Object> getVariantDebugInfo() {
+    return variantDebugInfo;
+  }
+
+  public static void setVariantDebugInfo(Map<String, Object> variantDebugInfo) {
+    VarCache.variantDebugInfo = variantDebugInfo;
```
Comment (rossenassenov) [inline AndroidSDKCore/src/main/java/com/leanplum/internal/VarCache.java]: check for null and assign empty map

**Pull request closed without merging**